core/src/main/scala/caliban/schema/Schema.scala

```diff
@@ -341,25 +341,43 @@ trait GenericSchema[R] extends SchemaDerivation[R] with TemporalSchema {
     ev2: Schema[RB, B]
   ): Schema[RA with RB, A => B]                                                                                        =
     new Schema[RA with RB, A => B] {
-      override def arguments: List[__InputValue] = {
-        val t = ev1.toType_(true)
-        t.inputFields.getOrElse(t.kind match {
-          case __TypeKind.SCALAR | __TypeKind.ENUM | __TypeKind.LIST =>
-            // argument was not wrapped in a case class, give it an arbitrary name
-            List(__InputValue("value", None, () => if (ev1.optional) t else makeNonNull(t), None))
-          case _                                                     => Nil
-        })
-      }
+      private val inputType                                                  = ev1.toType_(true)
```

Comment:
Can you make this one lazy? It might cause issues with recursive types if we call `toType_` while building the schema.